contracts/AppCoinsCreditsBalance.sol

```diff
@@ -0,0 +1,78 @@
+pragma solidity ^0.4.24;
+
+import "./AppCoins.sol";
+import "./Base/Whitelist.sol";
+
+contract AppCoinsCreditsBalance is Whitelist {
+
+    // AppCoins token
+    AppCoins private appc;
+
+    // balance proof
+    bytes balanceProof;
+
+    // balance
+    uint balance;
+
+    event BalanceProof(bytes _merkleTreeHash);
+    event Deposit(uint _amount);
+    event Withdraw(uint _amount);
+
+    constructor(
+        address _addrAppc
+    )
+    public
+    {
+        appc = AppCoins(_addrAppc);
+    }
+
+    /**
+    *@notice Get the balance
+    *@dev
+    *    returns the balance
+    *@return {"balance" : "balance"}
+    */
+    function getBalance() public view returns(uint256) {
+        return balance;
+    }
+
+    /**
+    *@notice Get the balance proof
+    *@dev
+    *    returns the balance proof
+    *@return {"balanceProof" : "balance proof"}
+    */
+    function getBbalanceProof() public view returns(bytes) {
+        return balanceProof;
+    }
+
+
+
+    function registerBalanceProof(bytes _merkleTreeHash)
+        public
+        onlyIfWhitelisted("registerBalanceProof",msg.sender){
+
+        balanceProof = _merkleTreeHash;
+
+        emit BalanceProof(_merkleTreeHash);
+    }
+
+    function depositFunds(uint _amount)
```

Comment:
Talking with @zepcp, registerBalanceProof can be an internal function called when depositFunds and withdrawFunds functions are called